modules/apps/portal-search/portal-search-web/src/test/java/com/liferay/portal/search/web/internal/custom/facet/display/context/CustomFacetDisplayContextTest.java

```diff
@@ -179,11 +182,209 @@ public void testOneTermWithPreviousSelection() throws Exception {
 		Assert.assertFalse(customFacetDisplayContext.isRenderNothing());
 	}
 
+	@Test
+	public void testOrderByTermFrequencyAscending() throws Exception {
+		List<TermCollector> termCollectors = _getTermCollectors(
+			new String[] {"alpha", "delta", "bravo", "charlie"},
+			new int[] {3, 4, 5, 6});
+
+		_setUpMultipleTermCollectors(termCollectors);
+
+		CustomFacetDisplayContext customSearchFacetDisplayContext =
+			_createDisplayContext(
+				"customDisplayCaption", "fieldToAggregate", StringPool.BLANK,
+				"count:asc");
+
+		List<BucketDisplayContext> bucketDisplayContexts =
+			customSearchFacetDisplayContext.getBucketDisplayContexts();
+
+		String nameFrequencyString = _buildNameFrequencyString(
+			bucketDisplayContexts);
+
+		Assert.assertEquals(
+			bucketDisplayContexts.toString(),
+			"alpha:3|delta:4|bravo:5|charlie:6", nameFrequencyString);
+
+		termCollectors = _getTermCollectors(
+			new String[] {"alpha", "delta", "bravo", "charlie"},
+			new int[] {4, 5, 5, 6});
+
+		_setUpMultipleTermCollectors(termCollectors);
+
+		customSearchFacetDisplayContext = _createDisplayContext(
+			"customDisplayCaption", "fieldToAggregate", StringPool.BLANK,
+			"count:asc");
+
+		bucketDisplayContexts =
+			customSearchFacetDisplayContext.getBucketDisplayContexts();
+
+		nameFrequencyString = _buildNameFrequencyString(bucketDisplayContexts);
+
+		Assert.assertEquals(
+			bucketDisplayContexts.toString(),
+			"alpha:4|bravo:5|delta:5|charlie:6", nameFrequencyString);
+	}
+
+	@Test
+	public void testOrderByTermFrequencyDescending() throws Exception {
+		List<TermCollector> termCollectors = _getTermCollectors(
+			new String[] {"alpha", "charlie", "bravo", "delta"},
+			new int[] {3, 4, 5, 6});
+
+		_setUpMultipleTermCollectors(termCollectors);
+
+		CustomFacetDisplayContext customFacetDisplayContext =
+			_createDisplayContext(
+				"customDisplayCaption", "fieldToAggregate", StringPool.BLANK,
+				"count:desc");
+
+		List<BucketDisplayContext> bucketDisplayContexts =
+			customFacetDisplayContext.getBucketDisplayContexts();
+
+		String nameFrequencyString = _buildNameFrequencyString(
+			bucketDisplayContexts);
+
+		Assert.assertEquals(
+			bucketDisplayContexts.toString(),
+			"delta:6|bravo:5|charlie:4|alpha:3", nameFrequencyString);
+
+		termCollectors = _getTermCollectors(
+			new String[] {"alpha", "delta", "bravo", "charlie"},
+			new int[] {4, 5, 5, 6});
+
+		_setUpMultipleTermCollectors(termCollectors);
+
+		customFacetDisplayContext = _createDisplayContext(
+			"customDisplayCaption", "fieldToAggregate", StringPool.BLANK,
+			"count:desc");
+
+		bucketDisplayContexts =
+			customFacetDisplayContext.getBucketDisplayContexts();
+
+		nameFrequencyString = _buildNameFrequencyString(bucketDisplayContexts);
+
+		Assert.assertEquals(
+			bucketDisplayContexts.toString(),
+			"charlie:6|bravo:5|delta:5|alpha:4", nameFrequencyString);
+	}
+
+	@Test
+	public void testOrderByTermValueAscending() throws Exception {
```

Comment:
Why are you testing order by Value Ascending and Value Descending if the user cannot order this facet by any of those?

Reply:
@gustavolimav why cant we order by value? i dont think theres any reason why we shouldnt be able to sort on value... let me know if im missing something, but i think those options need to be added to the config UI

Reply:
@BryanEngler In this story we only have two options to ordering, by frequency ascending and frequency descending. We dont have the value, like the others.

Reply:
@felipelorenz ya i think the story is wrong. Could check with @lipusz, but i think all 4 options should be there